#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From af3b1901549baa8fbe8140d9fa75a4a2b7a77a7e Mon Sep 17 00:00:00 2001
From: Werner Koch <wk@gnupg.org>
Date: Tue, 7 Sep 2021 08:57:44 +0200
Subject: [PATCH GnuPG] agent: Fix segv in GET_PASSPHRASE (regression)

* agent/command.c (cmd_get_passphrase): Do not deref PI.  PI is always
NULL.
--

Fixes-commit: b89b1f35c29ceaebe39b31444936aa66c9297f2c
GnuPG-bug-id: 5577
---
 agent/command.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/agent/command.c b/agent/command.c
index 5e2dbc809..3e074443a 100644
--- a/agent/command.c
+++ b/agent/command.c
@@ -1955,7 +1955,7 @@ cmd_get_passphrase (assuan_context_t ctx, char *line)
                                  opt_qualbar, cacheid, CACHE_MODE_USER, NULL);
       xfree (entry_errtext);
       entry_errtext = NULL;
-      is_generated = !!(pi->status & PINENTRY_STATUS_PASSWORD_GENERATED);
+      is_generated = 0;

       if (!rc)
         {
--
2.33.0
